### PR TITLE
Update html-test ref, the variables got renamed.

### PR DESCRIPTION
**Pull request by dfordivam**
PR corresponding to ghc/!5555 which caused a change in ModDetails in case of NoBackend. Now the initModDetails is used to recreate the ModDetails from interface and in-memory ModDetails is not used.

Ref: https://gitlab.haskell.org/ghc/ghc/-/merge_requests/5555

### CODE REVIEW

**Comment by Kleidukos** [comment]
@dfordivam Hi, and thank you for this PR! Shall I wait for your GHC MR to land before merging this one, or do you need it ASAP in the submodule?

**Comment by dfordivam** [comment]
So I am doing this kind of thing first time and just now reading the https://gitlab.haskell.org/ghc/ghc/-/wikis/working-conventions/git/submodules#making-changes-to-ghcs-submodules

I guess I need to first get the ghc side of MR to be in. So I will convert this to a draft..

**Comment by dfordivam** [comment]
@Kleidukos I think this is good for merge now. 

!5555 has finally passed the CI and if I understand correctly it will be put in marge queue once this lands in.

**Review by Kleidukos — APPROVED** [review]
LGTM.

**Pull request merged**
